### PR TITLE
Jay nn train

### PR DESCRIPTION
**Pull request by jayliu223**
This PR consists of script for preprocessing dataset (sentiment140) as well as ML model implementation through Tensorflow (since these steps are cumulative, I ended up lumping them all together).

The dataset is truncated so that only text and label columns are present. 50,000 most frequent words are tokenized and used to generate the embedding layer. A glimpse of the dataset:
image

0 = negative, 1 = positive.

Due to the absence of neutral label, binary cross entropy is selected as the loss function. 

NN parameters:
image

Test/train split is 0.9/0.1 (90% of the data is used for training, and 10% for validation). After 30 epochs, validated accuracy is at 82%.

Status for a demo epoch (reloaded ML model):
image

A note on LFS files - git threw errors due to many of the datasets/models being larger than 50/100MB in size, and git LFS was used to store data. 

Overall, I am not happy with the performance of the model. Due to the training dataset labeling, the model is incapable of accurately predicting neutral tweets, though it excels somewhat at predicting negative and positive sentiments.

My next step would be to find additional datasets and train the model using loss functions more appropriate for multiple classes. The current model will give us something to play with in the mean time. 

Other random edits:

Documentations have been moved under docs. Jupyter notebooks have been removed. Requirements have been updated (warning, some packages are not platform-universal).

If using Nvidia GPU, makes sure Python version is 3.9 or lower. You will also need CUDA toolkit and cudnn. Read more here: 
https://www.nvidia.com/en-sg/data-center/gpu-accelerated-applications/tensorflow/

If you're using Apple M1/M2, check this link:
https://developer.apple.com/metal/tensorflow-plugin/

Final note: git LFS is a mystery I have yet to solve. Note that some of the key files will be kept as pointers, and code may exhibit unexpected behavior until actual files are downloaded.

Download git LFS here: https://git-lfs.com/
Use git lfs pull to pull down large files.

### CODE REVIEW

**Review by MahreenAsghar — APPROVED** [review]
Good Job on rearranging the files and the ML code looks promising too!
